gcloud-java-dns/src/main/java/com/google/gcloud/dns/ZoneInfo.java

```diff
@@ -240,7 +266,7 @@ public Builder toBuilder() {
   }
 
   static ZoneInfo fromPb(com.google.api.services.dns.model.ManagedZone pb) {
-    Builder builder = new Builder(pb.getName());
+    Builder builder = new BuilderImpl(pb.getName());
```

Comment:
Also, you need to change the equals method to be symmetric.
See Storage.BucketInfo#equals as example.